### PR TITLE
Feature/423 admin dashboard display user info

### PR DESCRIPTION
**Pull request by HugoIneat**
- add user token
- manage copy to clipboard for api token
- add SMS consumption

### CODE REVIEW

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 1]
```diff
@@ -109,11 +141,14 @@ const AdminDashboard = () => {
                                 </Grid>
                                 <Divider />
                                 <ListItem className={classes.list}>
+                                    {
+                                    details &&
                                     <ListItemText
-                                        primary={`250 SMS ${t('consumed')} ${t(
+                                        primary={`${ getSMSUsage() !== undefined ? getSMSUsage() : '?' } SMS ${t('consumed')} ${t(
```
Comment (ntchou) [inline src/pages/AdminDashboard/AdminDashboard.tsx]:
```suggestion
                                        primary={`${ getSMSUsage() || '?' } SMS ${t('consumed')} ${t(
```

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 2]
```diff
@@ -109,11 +141,14 @@ const AdminDashboard = () => {
...
                                             'on'
-                                        )} 500 ${t('available')}`}
+                                        )} ${ getSMSQuota() !== undefined ? getSMSQuota() : '?' } ${t('available')}`}
```
Comment (ntchou) [inline src/pages/AdminDashboard/AdminDashboard.tsx]: same as above

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 3]
```diff
@@ -19,9 +19,33 @@ import ListItemText from '@material-ui/core/ListItemText'
...
 
 const AdminDashboard = () => {
-    const { t } = useTranslation('dashboard')
+    const dispatch = useDispatch()
+    const { t } = useTranslation('dashboard');
+    const { userId, token, details } = useSelector(selectUser)
+
+    useEffect(() => {
+
+        const getUserDetails = async () => {
```
Comment (ntchou) [inline src/pages/AdminDashboard/AdminDashboard.tsx:38]: i  think it would be better to have an action that combines all this.

**Review by MattiaPrimavera — APPROVED** [review]

File: src/pages/AdminDashboard/AdminDashboard.tsx [hunk 4]
```diff
@@ -53,6 +77,10 @@ const AdminDashboard = () => {
         openPremiumVideoCall(history)
     }
 
+    const getSMSUsage = () => details?.usage?.sentSMSs
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/AdminDashboard.tsx:80]: could be converted to selectors, 
getUserUsageSelector = state => getUserState(state).usage for example.

File: src/services/api.ts [hunk 5]
```diff
@@ -62,6 +62,21 @@ export class Api {
         })
     }
 
+    async getUserDetails(userId) : Promise<any> {
```
Comment (HugoGresse) [inline src/services/api.ts:65]: we can probably do beter typin than Promise<any> 
Something like Promise<User> maybe?

Comment (ntchou) [reply src/services/api.ts:65]: Yes instead of `any` `unknown`. Ideally share types between `functions` and `front`

**Pull request merged**